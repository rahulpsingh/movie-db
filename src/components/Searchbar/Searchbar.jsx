import React, { useState, useCallback } from "react";
import styled from "styled-components";
import search from "../../images/search.svg";
import getMovies from "../../utils/getMovies";
import debounce from "../../utils/debounceFunction";

const FormWrapper = styled.div`
  display: flex;
  flex-direction: row-reverse;
  border: 1px solid lightgrey;
  border-radius: 3px;
  justify-content: space-between;
  align-content: center;
  position: relative;
`;

const Input = styled.input`
  padding: 0px 35px 0px 35px;
  outline: none;
  border: none;

  &::placeholder {
    color: rgb(146, 146, 146);
  }
`;

const SearchIcon = styled.img`
  position: absolute;
  height: 50%;
  top: 25%;
  left: 8px;
`;

const SearchBar = ({ setSearchResults }) => {
  const [input, setInput] = useState("");

  const getSearch = async (query) => {
    try {
      const res = await getMovies("search", query);
      setSearchResults(res);
    } catch (error) {
      console.error(error);
    }
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const debounceDropdown = useCallback(
    debounce((nextValue) => getSearch(nextValue), 400),
    []
  );

  const handleInput = (e) => {
    if (e.target.value === "") {
      setSearchResults([]);
    }
    const query = e.target.value.split(" ").join("+");
    debounceDropdown(query);
  };

  return (
    <FormWrapper>
      <Input
        onInput={(e) => setInput(e.target.value)}
        value={input}
        type="text"
        onChange={handleInput}
        placeholder="Search for a movie"
      />
      <SearchIcon className="search-icon" src={search} alt="search-img" />
    </FormWrapper>
  );
};

export default SearchBar;
