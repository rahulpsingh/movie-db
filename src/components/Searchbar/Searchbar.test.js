import { render } from "@testing-library/react";
import Searchbar from "./Searchbar";

describe("Searchbar", () => {
  it("should render the Searchbar", () => {
    const { container } = render(<Searchbar toggle={() => {}} movieInfo={{}} visible />);
    expect(container).toMatchSnapshot();
  });
});
