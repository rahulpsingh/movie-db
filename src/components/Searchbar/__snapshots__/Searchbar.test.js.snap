// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar should render a modal 1`] = `
<div>
  <div
    class="sc-bdnxRM kARabk"
  >
    <input
      class="sc-gtsrHT iVZvUU"
      placeholder="Search for a movie"
      type="text"
      value=""
    />
    <img
      alt="search-img"
      class="sc-dlnjwi kObtID search-icon"
      src="search.svg"
    />
  </div>
</div>
`;

exports[`Searchbar should render the Searchbar 1`] = `
<div>
  <div
    class="sc-bdnxRM kARabk"
  >
    <input
      class="sc-gtsrHT iVZvUU"
      placeholder="Search for a movie"
      type="text"
      value=""
    />
    <img
      alt="search-img"
      class="sc-dlnjwi kObtID search-icon"
      src="search.svg"
    />
  </div>
</div>
`;
